src/etcd/lease.cr

```diff
@@ -6,37 +6,45 @@ class Etcd::Lease
   def initialize(@client = Etcd::Client.new)
   end
 
-  # Requests a lease
-  # ttl   ttl of granted lease                            Int64
-  # id    id of 0 prompts etcd to assign any id to lease  UInt64
-  def grant(ttl : Int64 = @ttl, id = 0)
-    Model::Grant.from_json(client.api.post("/lease/grant", {TTL: ttl, ID: 0}).body)
+  # /kv/lease/leases
+  # /lease/leases
+  # Queries for all existing leases in an etcd cluster
+  def leases
+    Model::LeasesArray.from_json(client.api.post("/kv/lease/leases").body).leases.map(&.id)
   end
 
-  # Requests persistence of lease.
-  # Must be invoked periodically to avoid key loss.
-  def keep_alive(id : Int64) : Int64?
-    Model::KeepAlive.from_json(client.api.post("/lease/keepalive", {ID: id}).body).result
-  rescue e : Exception
-    raise e unless e.message == "JSON key not found: TTL"
-    nil
+  # /kv/lease/revoke
+  # /lease/revoke
+  # Revokes an etcd lease
+  # id  Id of lease  Int64
+  def revoke(id : Int64)
+    # To get header: Etcd::Model::WithHeader.from_json(response.body)
+    client.api.post("/kv/lease/revoke", {ID: id}).success?
   end
 
+  # /kv/lease/timetolive
+  # /lease/timetolive
   # Queries the TTL of a lease
   # id            id of lease                         Int64
   # query_keys    query all the lease's keys for ttl  Bool
   def timetolive(id : Int64, query_keys = false)
     Model::TimeToLive.from_json(client.api.post("/kv/lease/timetolive", {ID: id, keys: query_keys}).body)
   end
 
-  # Revokes an etcd lease
-  # id  Id of lease  Int64
-  def revoke(id : Int64)
-    client.api.post("/kv/lease/revoke", {ID: id}).success?
+  # /lease/grant
+  # Requests a lease
+  # ttl   ttl of granted lease                            Int64
+  # id    id of 0 prompts etcd to assign any id to lease  UInt64
+  def grant(ttl : Int64 = @ttl, id = 0)
+    Model::Grant.from_json(client.api.post("/lease/grant", {TTL: ttl, ID: 0}).body)
   end
 
-  # Queries for all existing leases in an etcd cluster
-  def leases
-    Model::LeasesArray.from_json(client.api.post("/kv/lease/leases").body).leases.map(&.id)
+  # /lease/keepalive
+  # Requests persistence of lease.
+  # Must be invoked periodically to avoid key loss.
+  def keep_alive(id : Int64) : Int64?
+    model = Model::KeepAlive.from_json(client.api.post("/lease/keepalive", {ID: id}).body)
+    raise Exception.new(model.error.not_nil!.to_s) if model.result.nil?
```

Comment:
I think raise an error with the information attached to it, and render it nicely.

i.e. `raise Etcd::Error.new(model.error)`